bitnami/kafka/templates/scripts-configmap.yaml

```diff
@@ -123,14 +123,14 @@ data:
     {{- end }}
     export EXTERNAL_ACCESS_PORT={{ .Values.externalAccess.service.ports.external }}
     {{- else if eq .Values.externalAccess.service.type "NodePort" }}
-    {{- if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
+    {{- if .Values.externalAccess.service.domain }}
+    export EXTERNAL_ACCESS_HOST={{ .Values.externalAccess.service.domain }}
+    {{- else if and .Values.externalAccess.service.usePodIPs .Values.externalAccess.autoDiscovery.enabled }}
     export EXTERNAL_ACCESS_HOST="${MY_POD_IP}"
     {{- else if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
```

Comment:
Sorry, didn't notice this before, but this line should use "and" too, to work like the previous "else if" line. Can you update this in the PR?

Reply:
{{- else if or .Values.externalAccess.service.useHostIPs .Values.externalAccess.autoDiscovery.enabled }}
did you say: this line use "and"? ,it can not do that. The reasons are as follows:
There may be a situation where both NodePort and autodiscovery are enabled, and it is generally expected to use hostIP + NodePort as the access point. According to Kubernetes conventions, useHostIPs is not explicitly enabled. If this is modified, it may cause confusion because users may be used to accessing with "curl ip" instead of the expected hostIP when NodePort and autodiscovery are enabled without useHostIPs being explicitly enabled.

here is the infor:
https://github.com/bitnami/charts/tree/main/bitnami/kafka#using-nodeport-services
Option A) Use random node ports using an initContainer that discover them automatically.

externalAccess.enabled=true
externalAccess.service.type=NodePort
externalAccess.autoDiscovery.enabled=true
serviceAccount.create=true
rbac.create=true